transcom/mymove
# MB-6075 Update move order updater service object to use transaction connection

## Description

> In the transaction block, when the move order is updated, it is not using a transaction. This means that if the update errors out then any changes in the block are still valid if unsuccessful.
> 
> This specific line: https://github.com/transcom/mymove/blob/master/pkg/services/move_order/move_order_updater.go#L115
> 
> AC: Convert update call to use transaction connection
> 
> Notes
> 
> We don’t need to extend query builder’s UpdateOne function to use transaction connections, let’s try and get this into the transaction in a way that uses pop directly.

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?
- I removed the query builder from the move_order_updater.go file since nothing references it is being used anymore
- Check out the test I added to cover the transaction block: https://github.com/transcom/mymove/pull/5548/files#diff-51a409f7eed58fd7ce8d8933133c97722aa932bc1f14f44f5adbd81afb4ae89dR98
- Service object MoveOrderUpdater is now renamed to `OrderUpdater` to accurately describe the object.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make server_run
make office_client_run
```

**To test**
1. Log in to office app as TOO, locally
2. Click on any move
3. Click on `View & edit orders`
4. Edit the SAC
5. Click on `Save`
6. Orders should have been successfully updated, still works

## Code Review Verification Steps

 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6075) for this change
